frontend/components/ManageStakingPage/StakingContractSection/index.tsx

```diff
@@ -101,91 +131,124 @@ export const StakingContractSection = ({
-        activeStakingProgramMeta?.canMigrateTo.includes(stakingProgram)));
-
-  const cantMigrateReason = useMemo(() => {
-    if (isSelected) {
-      return 'Contract is already selected';
+  const migrateValidation = useMemo<{
+    canMigrate: boolean;
+    reason?: string | ReactNode;
+  }>(() => {
+    // loading requirements
+    if (!isBalanceLoaded) {
+      return { canMigrate: false, reason: <Text>Loading balance...</Text> };
     }
-
-    if (!activeStakingProgramMeta?.canMigrateTo.includes(stakingProgram)) {
-      return 'Migration not supported for this contract';
+    if (!isStakingContractInfoLoaded) {
+      return {
+        canMigrate: false,
+        reason: <Text>Loading staking contract information...</Text>,
+      };
     }
 
-    if (!hasEnoughRewards) {
-      return 'No available rewards';
+    // general requirements
+    if (isSelectedOrUnstakedDefault) {
+      return {
+        canMigrate: false,
+        reason: <Text>Contract is already selected</Text>,
+      };
     }
 
-    if (!isBalanceLoaded) {
-      return 'Loading balance...';
+    if (!hasEnoughRewards) {
+      return { canMigrate: false, reason: <Text>No available rewards</Text> };
     }
-
     if (!hasEnoughSlots) {
-      return 'No available staking slots';
+      return {
+        canMigrate: false,
+        reason: <Text>No available staking slots</Text>,
+      };
     }
-
-    if (!isServiceStakedForMinimumDuration && activeStakingContractInfo) {
-      return (
-        <CountdownUntilMigration
-          activeStakingContractInfo={activeStakingContractInfo}
-        />
-      );
+    if (!hasEnoughOlasToMigrate) {
+      return {
+        canMigrate: false,
+        reason: <Text>Insufficient OLAS to migrate</Text>,
+      };
+    }
+    // stopped service requirements
+    if (
+      [
+        DeploymentStatus.DEPLOYED,
+        DeploymentStatus.DEPLOYING,
+        DeploymentStatus.STOPPING,
+      ].some((status) => status === serviceStatus) // allow nil serviceStatus to pass, so first-timers can migrate
+    ) {
+      return {
+        canMigrate: false,
+        reason: <Text>Pearl is currently running</Text>,
+      };
     }
 
-    if (!hasEnoughOlasToMigrate) {
-      return `Insufficient OLAS to migrate, ${minimumOlasRequiredToMigrate} OLAS required in total.`;
+    // active staking program requirements
+    if (!isStakingContractInfoLoaded) {
+      return {
+        canMigrate: false,
+        reason: <Text>Loading staking contract information...</Text>,
+      };
     }
 
-    // App version compatibility not implemented yet
-    // if (!isAppVersionCompatible) {
-    //   return 'Pearl update required to migrate';
-    // }
+    // user is not actively staked, allow migration
+    if (activeStakingProgram === null) {
+      return { canMigrate: true };
+    }
 
-    if (serviceStatus === DeploymentStatus.DEPLOYED) {
-      return 'Pearl is currently running, turn it off before switching';
+    // USER IS STAKED
```

Comment:
is this comment correct? it says "is staked" while after goes "Service is not staked"
maybe can remove the comment at all?